src/commands.cpp

```diff
@@ -8201,7 +8223,45 @@ bool CommandBackupPut::procresult(Result r)
         e = r.errorOrOK();
     }
 
-    LOG_debug << "backup put result: " << error(e) << " " << backupId;
+    // automatically add the backup name to the corresponding user attribute
+    std::string key {Base64Str<MegaClient::BACKUPHANDLE>(backupId)};
+    std::string value = mBackupName;
+    string_map attrMap;
+    attrMap[key] = value;
+    attr_t attrtype = ATTR_BACKUP_NAMES;
+
+    std::unique_ptr<TLVstore> tlv;
+
+    User *ownUser = client->finduser(client->me);
+    const std::string *oldValue = ownUser->getattr(attrtype);
+    if (!oldValue)  // attr doesn't exist -> create it
+    {
+        tlv.reset(new TLVstore());
+        tlv->set(key, value);
+
+        // serialize and encrypt the TLV container
+        std::unique_ptr<std::string> container(tlv->tlvRecordsToContainer(client->rng, &client->key));
+        client->putua(attrtype, (byte *)container->data(), unsigned(container->size()));
+    }
+    else if (!ownUser->isattrvalid(attrtype)) // not fetched yet or outdated
+    {
+        LOG_err << "Failed to set backup name for backup id : " << backupId;
+    }
+    else
+    {
+        tlv.reset(TLVstore::containerToTLVrecords(oldValue, &client->key));
+
+        if (User::mergeUserAttribute(attrtype, attrMap, *tlv.get()))
+        {
+            // serialize and encrypt the TLV container
+            std::unique_ptr<std::string> container(tlv->tlvRecordsToContainer(client->rng, &client->key));
+            client->putua(attrtype, (byte *)container->data(), unsigned(container->size()));
+        }
+        else
+        {
+            LOG_err << "Failed to merge with existing backup names with the new one for backup id: " << backupId;
+        }
+    }
```

Comment:
this shouldn't be happening in case backupId is UNDEF, shouldn't it?

Reply:
Definitely. Good catch. Fixed